src/test.js

```diff
@@ -30,13 +30,24 @@ export default function Test( settings ) {
 	++Test.count;
 
 	this.expected = null;
-	extend( this, settings );
 	this.assertions = [];
 	this.semaphore = 0;
 	this.module = config.currentModule;
 	this.stack = sourceFromStacktrace( 3 );
 	this.steps = [];
 
+	// If a module is skipped, all its tests and the tests of the child suites
+	// should be treated as skipped even if they are defined as `only` or `todo`.
```

Comment:
>I'm unsure if we want to do this for only tests. I can see a scenario where a module is skipped, but then they want to work on an individual test without re-enabling the whole suite.

Sorry if I misunderstood something, but does that mean the `only` test should be the only one to run?
In that case, no need to skip the whole module, `QUnit` will skip all tests and only run the `only` test.



> I tend to think there needs to be some order of precedence for the modifiers. I think only > skip > todo > test. Does that make sense?

I see it like  this: `skip > only > todo > test`. Saying so, means that even if a `only` test is found inside a skipped module it should be skipped as that what the user expect to happen.

Also, Judging by the order, If a skipped test is inside a `todo` module, it should be left intact, right ?